lib/surface/components/link.ex

```diff
@@ -60,32 +60,33 @@ defmodule Surface.Components.Link do
   slot default
 
   def render(assigns) do
-    valid_label!(assigns)
+    unless assigns[:default] || assigns[:label] || Keyword.get(assigns.opts, :label) do
+      raise ArgumentError, "<Link /> requires a label prop or contents in the default slot"
+    end
+
     to = valid_destination!(assigns.to, "<Link />")
+    events = events_to_opts(assigns)
+    opts = link_method(assigns.method, to, assigns.opts)
+    assigns = assign(assigns, to: to, opts: events ++ opts)
```

Comment:
👍

Reply:
We need to remember to use `assign/2..3` and friends in all our components. No need to do that in this PR though. Only the ones you already have to touch.

Reply:
I'll open a different PR to migrate all remaining components to the new `assign/2..3` way.